pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/adminoperation/UpdateKitToLegacyIdService.java

```diff
@@ -155,10 +159,13 @@ private static void changeDataInEs(String currentCollaboratorSampleId, String ne
      * */
     private static boolean existsKitRequestWithCollaboratorSampleId(String collaboratorSampleId, String ddpParticipantId,
                                                             KitRequestDao kitRequestDao) {
-        KitRequestShipping kitRequestShipping = kitRequestDao.getKitRequestForCollaboratorSampleId(collaboratorSampleId);
-        if (kitRequestShipping == null) {
+        List<KitRequestShipping> kitRequestShippingList = kitRequestDao.getKitRequestForCollaboratorSampleId(collaboratorSampleId);
+        if (kitRequestShippingList == null || kitRequestShippingList.isEmpty()) {
             return false;
         }
-        return StringUtils.isNotBlank(ddpParticipantId) ? ddpParticipantId.equals(kitRequestShipping.getDdpParticipantId()) : true;
+        if (kitRequestShippingList.size() > 1) {
+            throw new DSMBadRequestException("Multiple kit requests found for collaborator sample ID %s".formatted(collaboratorSampleId));
```

Comment:
Likewise, this check should be done earlier to avoid leaving ES out of sync with the DB.

Reply:
This method is only called before any db or ES update is made. It's to see if a kit with this sample id exists or not. Since sample id is supposed to be unique, but there is no unique constraint on it in db, if there are more than one samples with this collaborator sample id this exception will stop the method from updating the db and es and returns an ERROR update log